plugins/spi/SPIOutput.cpp

```diff
@@ -187,60 +186,79 @@ SPIOutput::SPIOutput(const UID &uid, SPIBackendInterface *backend,
-
-  personalities.insert(personalities.begin() + PERS_P9813_INDIVIDUAL - 1,
-                       Personality(m_pixel_count * P9813_SLOTS_PER_PIXEL,
-                                   "P9813 Individual Control"));
-  personalities.insert(personalities.begin() + PERS_P9813_COMBINED - 1,
-                      Personality(P9813_SLOTS_PER_PIXEL,
-                                  "P9813 Combined Control"));
-
-  personalities.insert(personalities.begin() + PERS_APA102_INDIVIDUAL - 1,
-                       Personality(m_pixel_count * APA102_SLOTS_PER_PIXEL,
-                                   "APA102 Individual Control"));
-
-  ola::rdm::SlotDataCollection::SlotDataList sd_APA102_COMBINED;
-  sd_APA102_COMBINED.push_back(
+  ola::rdm::SlotDataCollection::SlotDataList sd_rgb_combined;
+  sd_rgb_combined.push_back(
       ola::rdm::SlotData::PrimarySlot(ola::rdm::SD_COLOR_ADD_RED, 0));
-  sd_APA102_COMBINED.push_back(
+  sd_rgb_combined.push_back(
       ola::rdm::SlotData::PrimarySlot(ola::rdm::SD_COLOR_ADD_GREEN, 0));
-  sd_APA102_COMBINED.push_back(
+  sd_rgb_combined.push_back(
       ola::rdm::SlotData::PrimarySlot(ola::rdm::SD_COLOR_ADD_BLUE, 0));
-  personalities.insert(personalities.begin() + PERS_APA102_COMBINED - 1,
-                      Personality(APA102_SLOTS_PER_PIXEL,
-                                  "APA102 Combined Control",
-       ola::rdm::SlotDataCollection(sd_APA102_COMBINED)));
+  ola::rdm::SlotDataCollection sdc_rgb_combined;
+  sdc_rgb_combined = ola::rdm::SlotDataCollection(sd_rgb_combined);
 
-  personalities.insert(personalities.begin() + PERS_APA102_PB_INDIVIDUAL - 1,
-                       Personality(m_pixel_count * APA102_PB_SLOTS_PER_PIXEL,
-                                   "APA102 Pixel Brightness Individ."));
-
-  ola::rdm::SlotDataCollection::SlotDataList sd_APA102_PB_COMBINED;
-  sd_APA102_PB_COMBINED.push_back(
+  ola::rdm::SlotDataCollection::SlotDataList sd_irgb_combined;
+  sd_irgb_combined.push_back(
       ola::rdm::SlotData::PrimarySlot(ola::rdm::SD_INTENSITY, 0));
-  sd_APA102_PB_COMBINED.push_back(
+  sd_irgb_combined.push_back(
       ola::rdm::SlotData::PrimarySlot(ola::rdm::SD_COLOR_ADD_RED, 0));
-  sd_APA102_PB_COMBINED.push_back(
+  sd_irgb_combined.push_back(
       ola::rdm::SlotData::PrimarySlot(ola::rdm::SD_COLOR_ADD_GREEN, 0));
-  sd_APA102_PB_COMBINED.push_back(
+  sd_irgb_combined.push_back(
       ola::rdm::SlotData::PrimarySlot(ola::rdm::SD_COLOR_ADD_BLUE, 0));
-  personalities.insert(personalities.begin() + PERS_APA102_PB_COMBINED - 1,
-                       Personality(APA102_PB_SLOTS_PER_PIXEL,
-                                   "APA102 Pixel Brightness Combined",
-      ola::rdm::SlotDataCollection(sd_APA102_PB_COMBINED)));
+  ola::rdm::SlotDataCollection sdc_irgb_combined;
+  sdc_irgb_combined = ola::rdm::SlotDataCollection(sd_irgb_combined);
+
+  personalities.insert(
+      personalities.begin() + PERS_WS2801_INDIVIDUAL - 1,
```

Comment:
You don't need to move this one down, but I understand you may want to make them all look the same.

Reply:
you understand my intention ;-)
i have made the style-change in an extra commit - so i could easily revoke this..
i like the 'list-style' this way most - but i know you mentioned the rule is normally 'all that fits in the same line'...

Reply:
That's a good enough reason for me! Rule is probably too strong, general style, but with enough exceptions you can do what you want. :smile: